packages/SwingSet/src/kernel/kernel.js

```diff
@@ -520,6 +523,50 @@ export default function buildKernel(kernelEndowments) {
     );
   }
 
+  let vatAdminVatMgr;
+
+  // Create a new vat, wait for the results, and notify the vat admin device.
+  function createVatAndNotify(buildFn, vatID) {
+    const setup = (syscall, state, helpers) => {
+      return helpers.makeLiveSlots(syscall, state, buildFn, helpers.vatID);
+    };
+
+    buildVatManager(vatID, `dynamicVat${vatID}`, setup, {});
+    // const kernelSlot = kernelKeeper.addKernelObject(vatID);
+    const vatSlot = makeVatSlot('object', true, 0);
+    const kernelrootObjSlot = addExport(vatID, vatSlot);
+    // const promiseSlot = kernelKeeper.addKernelPromise(vatID);
+    const newVatManager = ephemeral.vats.get(vatID).manager;
+    newVatManager.mapKernelSlotToVatSlot(kernelrootObjSlot);
+    vatAdminVatMgr.mapKernelSlotToVatSlot(kernelrootObjSlot);
```

Comment:
hm, these three lines feel out-of-place.

I think `newVatManager.mapKernelSlotToVatSlot(kernelrootObjSlot)` is accomplished inside `addExport`: it will ensure that both directions of the vat c-list table are present, allocating them if necessary.

And `vatAdminVatMgr...` should be accomplished when the `newVatCallback` message we enqueue here gets delivered. At that point, the message slots will be mapped back into vat slots (for the `vatAdminVat`), so we don't need to pre-map it here.

I think that might mean we can get rid of the `vatAdminVatMgr` variable too, if it's not used by anything else. We need to remember the vat-admin vat's ID, for the `queueToExport` below, but not the its VatManager.